internal/controller/warehouses/images.go

```diff
@@ -129,16 +129,18 @@ func imageDiscoveryLogFields(sub kargoapi.ImageSubscription) []any {
 		"platformConstrained", sub.Platform != "",
 	}
 	switch sub.ImageSelectionStrategy {
-	case kargoapi.ImageSelectionStrategySemVer, kargoapi.ImageSelectionStrategyDigest:
-		f = append(
-			f,
-			"semverConstraint", sub.SemverConstraint,
-		)
+	case kargoapi.ImageSelectionStrategySemVer:
+		if sub.Constraint != "" {
+			f = append(f, "constraint", sub.Constraint)
+		} else {
+			f = append(f, "semverConstraint", sub.SemverConstraint) //nolint:staticcheck
```

Comment:
These k/v pairs are adding context to all the discovery logging that's about to happen. If everything else has been done correctly, the discovery process is only aware of a constraint and not really aware of _how_ that constraint came to be set. So I think I would probably use `constraint` as the key here instead of `semverConstraint`.

```suggestion
			f = append(f, "constraint", sub.SemverConstraint) //nolint:staticcheck
```